### PR TITLE
Fix use of versioned github action

### PR DESCRIPTION
**Pull request by karl-johan-grahn**
The README would need to be updated too because tagging the GitHub action runner won't work for earlier versions that this PR will eventually be part of - earlier versions will always use `main`. Updating the README has to be done after this PR has been merged though, since we don't know the version this fix will be part of.

Fixes: #3761
Related: https://github.com/orgs/community/discussions/49245
Related: https://github.com/actions/runner/issues/2473

### CODE REVIEW

**Comment by ssbarnea** [comment]
@karl-johan-grahn Thanks for working on this. Sadly, you cannot expand environment variables inside GHA expressions `${{ ... }}` as they will give syntax error.

File: action.yml [hunk 1]
```diff
@@ -25,8 +25,10 @@ runs:
     # https://github.com/actions/setup-python/issues/361
     - name: Generate .git/ansible-lint-requirements.txt
       shell: bash
+      env:
+        GH_ACTION_REF: ${{ github.action_ref }}
       run: |
-        wget --output-file=.git/ansible-lint-requirements.txt https://raw.githubusercontent.com/ansible/ansible-lint/${{ github.action_ref || 'main' }}/.config/requirements-lock.txt
+        wget --output-file=.git/ansible-lint-requirements.txt https://raw.githubusercontent.com/ansible/ansible-lint/${{ $GH_ACTION_REF || 'main' }}/.config/requirements-lock.txt
```
[the remark this answers is not in the document]
Comment (karl-johan-grahn) [reply action.yml:31]: Not sure I follow, how do you mean with `github.action_path` and copy?

File: action.yml [hunk 2]
```diff
@@ -25,8 +25,10 @@ runs:
     # https://github.com/actions/setup-python/issues/361
     - name: Generate .git/ansible-lint-requirements.txt
       shell: bash
+      env:
+        GH_ACTION_REF: ${{ github.action_ref || 'main' }}
       run: |
-        wget --output-file=.git/ansible-lint-requirements.txt https://raw.githubusercontent.com/ansible/ansible-lint/${{ github.action_ref || 'main' }}/.config/requirements-lock.txt
+        wget "--output-file=.git/ansible-lint-requirements.txt https://raw.githubusercontent.com/ansible/ansible-lint/$GH_ACTION_REF/.config/requirements-lock.txt"
```
[the remark this answers is not in the document]
Comment (karl-johan-grahn) [reply action.yml]: No worries, got the same error locally and realized the same fix was needed

File: action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (karl-johan-grahn) [reply action.yml]: Thanks for the insight, I'll leave it as is for now though

**Comment by karl-johan-grahn** [comment]
@ssbarnea please review and any plans to merge this soon? 🙇

**Comment by karl-johan-grahn** [comment]
Thanks @ChristopherHX hoping for review from @ssbarnea soon. We use Renovate and it is based on tags so I'd like this to work to get automatic dependency updates for this action.

**Comment by ssbarnea** [comment]
I am looking at the current code and I don't really understand what is fixing. Basically it introduced use of environment variables instead of direct expansion using github own expressions. The result would be identical, so what is the catch?

**Comment by karl-johan-grahn** [comment]
Nothing has changed, the issue this PR is still attempting to address is https://github.com/ansible/ansible-lint/issues/3761, which persists

**Comment by karl-johan-grahn** [comment]
Is the action nested for this repo's workflow?

**Comment by karl-johan-grahn** [comment]
At least it will be one step better than now

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**